src/wallet/wallet.cpp

```diff
@@ -1177,16 +1178,22 @@ int64_t CWalletTx::GetTxTime() const
     return n ? n : nTimeReceived;
 }
 
-int64_t CWalletTx::GetComputedTxTime() const
+/**
+ * Update smart timestamp for a transaction being added to the wallet.
+ *
+ * Logic:
+ * - If the transaction is not yet part of a block, assign its timestamp to the current time.
+ * - Else assign its timestamp to the block time
+ */
+void CWalletTx::UpdateTimeSmart()
 {
-    LOCK(cs_main);
-    if (ContainsZerocoins()) {
-        if (IsInMainChain())
-            return mapBlockIndex.at(hashBlock)->GetBlockTime();
-        else
-            return nTimeReceived;
+    nTimeSmart = nTimeReceived;
+    if (!hashBlock.IsNull()) {
+        if (mapBlockIndex.count(hashBlock)) {
+            nTimeSmart = mapBlockIndex[hashBlock]->GetBlockTime();
```

Comment:
lets not use the `[]` operator here and use `mapBlockIndex.at(hashBlock)->GetBlockTime();` instead. Even though this is guarded by the above `if` check, probably best to stick to never using the `[]` operator when reading from a map. (https://github.com/PIVX-Project/PIVX/blob/master/doc/developer-notes.md#c-data-structures)

Reply:
Yes, even though this line was already like this in the previous `ComputeTimeSmart` (and, as you pointed out, makes no difference, given the `if` check), it makes sense to fix it here, for consistency.
Updated.